Wasm/Engine.lean

```diff
@@ -83,25 +85,33 @@ def mkStore (m : Module) : Store m :=
   Store.mk $ instantiateFs m
 
 def funcByName (s : Store m) (x : String) : Option $ FunctionInstance m :=
-  match s.func.filter (fun f => f.name == .some x) with
+  match s.func.filter (fun f => f.export_ == .some x) with
   | y :: [] => .some y
   | _ => .none
 
 def fidByName (s : Store m) (x : String) : Option Nat :=
   funcByName s x >>= pure ∘ FunctionInstance.index
 
-def paramTypecheck (x : Local) (y : StackEntry) :=
+def stackEntryTypecheck (x : Type') (y : StackEntry) :=
   match y with
   | .num nn => match nn with
-    | .i n => match x.type with
+    | .i n => match x with
       | .i 32 => n.bs == 32
       | .i 64 => n.bs == 64
       | _ => false
-    | .f n => match x.type with
+    | .f n => match x with
       | .f 32 => n.bs == 32
       | .f 64 => n.bs == 64
       | _ => false
 
+-- Checks that the given numerical values correspond to the given
+-- types both in the type of each respective value and in length.
+def resultsTypecheck : List Type' → List StackEntry → Bool
+  | [], [] => true
+  | t :: ts, e :: es =>
```

Comment:
I finally understand this syntax now, thank you.
Now let's see if I remember it a week after.